src/org/benf/cfr/reader/entities/Field.java

```diff
@@ -179,11 +182,62 @@ public void collectTypeUsages(TypeUsageCollector collector) {
         collector.collectFromT(attributes.getByName(AttributeRuntimeInvisibleTypeAnnotations.ATTRIBUTE_NAME));
     }
 
+    private List<AnnotationTableEntry> getRecordComponentAnnotations(ClassFile owner, List<AnnotationTableEntry> fieldAnnotations) {
+        // Note: Don't have to consider TYPE_USE annotations because they are implicitly propagated
+        // to field and already dumped as part of regular field annotation handling
+        
+        List<AnnotationTableEntry> componentAnnotations = null;
+
+        AttributeRecord recordAttribute = owner.getAttributes().getByName(AttributeRecord.ATTRIBUTE_NAME);
+        List<Attribute> componentAttributes;
+        if (recordAttribute != null && (componentAttributes = recordAttribute.getRecordComponentAttributes(fieldName)) != null) {
+            componentAnnotations = MiscAnnotations.BasicAnnotations(new AttributeMap(componentAttributes));
+        }
+
+        if (componentAnnotations == null) {
+            // Return all annotations which were propagated from component to field
+            return fieldAnnotations;
+        } else if (fieldAnnotations == null) {
+            return componentAnnotations;
+        } else {
+            // Create copy because original list might not be modifiable
+            componentAnnotations = ListFactory.newList(componentAnnotations);
+
+            // First collect the type of all annotations with target RECORD_COMPONENT
+            Set<JavaTypeInstance> componentAnnotationTypes = Functional.mapToSet(componentAnnotations, new UnaryFunction<AnnotationTableEntry, JavaTypeInstance>() {
+                @Override
+                public JavaTypeInstance invoke(AnnotationTableEntry arg) {
+                    return arg.getClazz();
+                }
+            });
+
+            // Then consider all annotations without RECORD_COMPONENT target, but with FIELD
+            // target which are implicitly propagated from component to field
+            // Annotations which are already in componentAnnotationTypes have both RECORD_COMPONENT
+            // and FIELD as target and therefore don't have to be added a second time
```

Comment:
In theory a custom class file might have annotations of the same type which differ between component and field, so this could be written more defensively. However, because a record cannot have any instance fields in source, such a class file could not be properly decompiled.